src/gui_x11.c

```diff
@@ -1282,6 +1282,17 @@ gui_mch_init_check(void)
 		cmdline_options, XtNumber(cmdline_options),
 		CARDINAL &gui_argc, gui_argv);
 
+# if defined(FEAT_FLOAT) && defined(LC_NUMERIC)
+    {
+	/* The call to XtOpenDisplay() may have set the locale from
+	 * the environment. Set LC_NUMERIC to "C" to make sure
+	 * that strtod() uses a decimal point, not a comma. */
+	char *p = setlocale(LC_NUMERIC, NULL);
+
+	if (p == NULL || strcmp(p, "C") != 0)
+	   setlocale(LC_NUMERIC, "C");
```

Comment:
I wondered whether calling setlocale(LC_NUMERIC, "C") unconditionally (i.e. without
checking for the current locale setting) would be fine. I suspect it would be fine, but
I saw that it was done that way in gui_gtk_x11.c so I did it in the same way.

Reply:
The one caveat is that if you want any "individual" locale settings to take hold, you should make sure that $LC_ALL is not set, because if it is, it will override all other locale settings. (OTOH, $LANG is not dangerous, since it only specifies a fallback value to be used for any unset locale categories.)

Best regards,
Tony.